src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -1474,6 +1585,13 @@ internal override void ForceComplete(SourceLocation locationOpt, CancellationTok
                         }
                         break;
 
+                    case CompletionPart.StartPropertyAccessorsBinding:
+                    case CompletionPart.FinishPropertyAccessorsBinding:
+                        {
+                            EnsureAccessorsBinding();
```

Comment:
>EnsureAccessorsBinding(); [](http://example.com/codeflow?start=28&length=25)

It feels like we will be binding accessors here for no good reason, even if they use the ```field``` keyword.

Reply:
@AlekseyTs Can you clarify further? The method name, `ForceComplete`, implied to me that it should literally force complete any missing `CompletionPart`s.

Reply:
>>It feels like we will be binding accessors here for no good reason, even if they use the ```field``` keyword.

>Can you clarify further? The method name, ForceComplete, implied to me that it should literally force complete any missing CompletionParts.

That is correct, the method must complete everything that requires completion. The question though is - why this is part of the completion? Why does it belong here? The completion is supposed to finish all pending analysis for the declaration. Binding accessors has nothing to do with the declaration. In previous commits we didn't do anything here and things were working just fine. I think that we are simply misusing the completion machinery so that we can distinguish three states of _lazyBackingFieldSymbol:
- null, but we don't know if we are going to create one;
- null, but we know that the field is not needed;
- not null (we know that the field is needed and we created one).
Because of that, we are forced to do an unnecessary work here 

Usually, we don't use any additional flags in order to be able to track three states like that. Instead, we use sentinel values to represent uninitialized/unknown state. See ```SourceMemberMethodSymbol._lazyIteratorElementType```, for example.